Add documentation on current existing forks (#19)
optskug/docs: README.md

        <strong>Long:</strong> Yes from community (original author) <br/> <br/>
        <strong>Notes - same as <code>sunnypilot/master-new</code> plus:</strong>
        <ul>
          <li>Includes a TSK keyboard with key caching</li>
          <li><code>commaai/master</code> and most forks based on it have unstable experimental mode due to the latest model. The model switcher in SP allows you to use old models with stable experimental mode</li>
          <li>Maintained as best effort. This may not always be up to date with latest SunnyPilot. Reach out to @Chrispypatt for requests to rebase his fork with SunnyPilot's rewrite.</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <code>optskug/SiennaFP</code> <br/> 
        (fork of FrogPilot) <br/>
        <a href="https://discord.com/invite/frogpilot">Frogpilot discord</a>
      </td>
      <td>
        <strong>Lat:</strong> Yes from community (from anrum) <br/>
        <strong>MADS:</strong> Yes from community (AOL from FP) <br/>
        <strong>Long:</strong> No <br/> <br/>
        <strong>Notes</strong>
        <ul>
          <li>anrum's old fork of FP and first fork to support TSK lateral!</li>

>           <li>Includes a TSK keyboard with key caching</li>

Reviewer: ```suggestion
          <li>Includes a TSK keyboard with key caching</li>
          <li>Includes auto key installer</li>
```

Then familiarize yourself with the communities through Discord for each fork you are looking to install.

If you acknowledge the warning above and are still looking to try a fork that supports TSK the following are available to install at your own risk. Keep in mind that these are community maintained and may not stay up to date:

<table>
    <tr>
        <td></td>
        <td>
          <strong>Lat:</strong> Lateral support
          <br>
          <strong>MADS:</strong> AOL / MADS / keep-lat-on-after-brakes
          <br>
          <strong>Long:</strong> Longitudinal support
          <br>
          <br>
          <strong>Notes:</strong>
        </td>
    </tr>
    <tr>
      <td><code>alexandresato/personal3</code>  <br/> 
        (a.k.a SatoPilot)
      </td>
      <td>
        <strong>Lat:</strong> Yes from upstream <br/>
        <strong>MADS:</strong> Yes from community (MADS from Spektor56) <br/>
        <strong>Long:</strong> Yes from community (from chrispypatt) <br/> <br/>
        <strong>Notes:</strong>
        <ul>
          <li>First fork to get long!</li>
          <li>Very quick stop-and-go response</li>

>           <li><code>`alexandresato/extract_secoc_key_btn</code> includes a TSK key extract button and is rebased with personal3 often.</li>

Reviewer: ```suggestion
          <li><code>alexandresato/extract_secoc_key_btn</code> includes a TSK key extract button and is rebased with <code>personal3</code> often.</li>
```

        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <code>chrispypatt/sunnypilot</code>  <br/> 
        (fork of SunnyPilot)
        <a href="https://discord.com/invite/sunnypilot">SunnyPilot discord</a>
      </td>
      <td>
        <strong>Lat:</strong> Yes from upstream <br/>
        <strong>MADS:</strong> Yes from community (MADS from SP) <br/>
        <strong>Long:</strong> Yes from community (original author) <br/> <br/>
        <strong>Notes - same as <code>sunnypilot/master-new</code> plus:</strong>
        <ul>
          <li>Includes a TSK keyboard with key caching</li>
          <li><code>commaai/master</code> and most forks based on it have unstable experimental mode due to the latest model. The model switcher in SP allows you to use old models with stable experimental mode</li>
          <li>Maintained as best effort. This may not always be up to date with latest SunnyPilot. Reach out to @Chrispypatt for requests to rebase his fork with SunnyPilot's rewrite.</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <code>optskug/SiennaFP</code> <br/> 
        (fork of FrogPilot) <br/>

>         <a href="https://discord.com/invite/frogpilot">Frogpilot discord</a>

Reviewer: ```suggestion
        <a href="https://discord.com/invite/frogpilot">FrogPilot discord</a>
```

      </td>
      <td>
        <strong>Lat:</strong> Yes from community (from anrum) <br/>
        <strong>MADS:</strong> Yes from community (AOL from FP) <br/>
        <strong>Long:</strong> No <br/> <br/>
        <strong>Notes</strong>
        <ul>
          <li>anrum's old fork of FP and first fork to support TSK lateral!</li>
          <li>Includes a TSK keyboard with key caching</li>
          <li>Includes auto key installer</li>
          <li>Uses an old AGNOS version. When downgrading, OP may get stuck in a registration loop. In this case, tap-tap-tap on the boot logo and reset the device to recover and then install again.</li>
        </ul>
      </td>
    </tr>
</table>

> [!WARNING]

> > `optskug/SiennaFP` is the only FrogPilot fork/branch with TSK support. Do not install the latest FP unless FrogGoesMoo confirmed that it supports TSK vehicles.

Reviewer: Frog just pushed a change in staging that claims to support TSK. Need to check.

Reviewer: If you want to merge, let's remove this warning

Reviewer: We can just take this out if you want to merge

Reviewer: https://discord.com/channels/1137853399715549214/1137905508217540699/1331156376336076840

Author: I think until we confirm it's working we should hold off on removing this

> If you are installing a fork not included in the list above, reach out to the fork owner and ask:

Reviewer: ```suggestion
If you are installing a fork not included in the list above, find the fork author and ask:
```

Go back to Step 4A-4 and then finish with Step 4A-5.

---
## Forks

### Which Fork Should I Use?

> > [!WARNING]
> > If you have to ask, you're not ready for forks.

Reviewer: ```suggestion
If you have to ask, you're not ready for forks.
```
Completely up to you on whether to take this change. With the caution box below, this seemed a bit redundant. Though on the other hand, plastering warning signs near a danger is also not a bad idea.

Author: I'm fine with this. I agree

Begin your research in Discord #custom-forks

Please do not ask about forks outside of that channel.

> Please start with `commaai/master-ci`. This is the latest upstream and supports lateral only. 

Reviewer: ```suggestion
If you're new, please start with `commaai/master-ci` and use it for two weeks. This is the latest official version with only lateral support. This will give you a good baseline to compare the other forks to.
```

> > [!WARNING]

Reviewer: ```suggestion
> [!CAUTION]
```

> 
> Some forks play a blood-curdling goat scream at max volume randomly.
> 
> The danger in using forks is not theoretical because it has full control over your car.
> 
> There's a real and present danger in using forks, even on the most popular forks.

If you acknowledge the warning above and are still looking to try a fork that supports TSK the following are available to install at your own risk. Please keep in mind that these are community maintained and may not stay up to date:

#### Forks with Long

* `alexandresato/personal3`
  * A.K.A SatoPilot, the first fork to support TSK long!
  * Very quick stop-and-go response
  * Also `alexandresato/extract_secoc_key_btn` includes a TSK key extract button and is rebased with personal3 often.
* `chrispypatt/sunnypilot`
  * Chrispypatt's fork of SunnyPilot's in-progress rewrite. 
  * Maintained as best effort. This may not always be up to date with latest SunnyPilot. Reach out to @Chrispypatt for requests to rebase his fork with SunnyPilot's rewrite.
  * Includes a TSK keyboard with key caching
  * Model switcher to easily switch between various models

#### Forks with Lateral Only

* `optskug/SiennaFP`

>   * @anrum's old fork of FP and first fork to support TSK lateral!

Reviewer: ```suggestion
  * anrum's old fork of FP and first fork to support TSK lateral!
```

  * Uses an old AGNOS version. When downgrading, OP may get stuck in a registration loop. In this case, tap-tap-tap on the boot logo and reset the device to recover and then install again.

> 

Reviewer: ```suggestion

> [!WARNING]
> `optskug/SiennaFP` is the only FrogPilot fork/branch with TSK support. Do not install the latest FP.
```

Go back to Step 4A-4 and then finish with Step 4A-5.

---
## Forks

### Which Fork Should I Use?

> If you have to ask, you're not ready for forks.

Reviewer 2: Maybe the caution should be moved up and this should end with "Here are the reasons why:"

Reviewer: Yeah it could use a cleanup. I think it's messy because chrispypatt is transcribing what I wrote in the past. Hey Chris, go wild and overhaul it in your own voice.

Author: I like all the points you had @calvinpark. I will move the caution up and will think for a bit on if there is a different way I want to organize these points or if there is anything I want to add.

Begin your research in Discord #custom-forks

Please do not ask about forks outside of that channel.

If you're new, please start with `commaai/master-ci` and use it for two weeks. This is the latest official version with only lateral support. This will give you a good baseline to compare the other forks to.

> [!CAUTION]
> Some forks will brick your C3X. \
> Most forks are not made for C3X because C3X is less than a year old.
> 
> Some forks contain banned code. \
> Using it will get you banned by the comma.ai company.
> 
> Some forks have nudgeless-lane-change. \
> Simply clicking the turn signal will move your car to the next lane. \
> Without any checks. \
> Yes, it will drive into the car next to you.
> 
> Some forks play a blood-curdling goat scream at max volume randomly.
> 

> > The danger in using forks is not theoretical because it has full control over your car.

Reviewer 2: I think maybe you should add that people should get familiar with the communities, especially their Discord communities and presence before using a fork. There may be strange changes and other errata that they might want to report. Features get a lot less testing and their interaction with SecOC/TSK vehicles may be suspect.

Reviewer: That's a good point

Reviewer: On a second thought, yes to getting familiar with their Discord (maybe we should put a link on the table).

As for the SecOC interaction, I think driving options are separate from SecOC. It's not like there's MADS and then a separate MADS for SecOC.

> 
> There's a real and present danger in using forks, even on the most popular forks.

If you acknowledge the warning above and are still looking to try a fork that supports TSK the following are available to install at your own risk. Please keep in mind that these are community maintained and may not stay up to date:

#### Forks with Long

> 

Reviewer 2: Should long should be documented as not as well tested for the moment and that users looking into forks might want to use a lat-only option and one that is more well supported like `master-new` on SP (if it works?).

Reviewer 2: Eventually, longitudinal support should be the norm and the distinction shouldn't need to be made.

Reviewer: It may make sense to write that the long support is from a community-made code that's not yet upstreamed, and thus its uses have not gone through the same amount of validations that lat support has

Reviewer: I mean for that matter, `optskug/SiennaFP`'s lat is also not from upstream. I guess it may make sense to note that there too

* `alexandresato/personal3`
  * A.K.A SatoPilot, the first fork to support TSK long!
  * Very quick stop-and-go response
  * Also `alexandresato/extract_secoc_key_btn` includes a TSK key extract button and is rebased with personal3 often.
* `chrispypatt/sunnypilot`
  * Chrispypatt's fork of SunnyPilot's in-progress rewrite. 
  * Maintained as best effort. This may not always be up to date with latest SunnyPilot. Reach out to @Chrispypatt for requests to rebase his fork with SunnyPilot's rewrite.
  * Includes a TSK keyboard with key caching
  * Model switcher to easily switch between various models

> #### Forks with Lateral Only

Reviewer 2: Does sunnypilot's `master-new` support lateral only as a byproduct of continuously syncing upstream?

Reviewer 2: On that note, what about MADS for SP? Or AOL for FP?  How do those interact with the various forks?

Reviewer: Yes on the first one.

Not sure if I follow the second question. MADS and AOL work on cars that the forks support. At this time FP doesn't support TSK and thus AOL is also not supported. MADS works.

Reviewer: Good idea. I wonder what SatoPilot calls AOL/MADS in his fork

Author: SatoPilot calls it MADS as well

Go back to Step 4A-4 and then finish with Step 4A-5.

---
## Forks
### Which Fork Should I Use?
> [!WARNING]
> If you have to ask, you're not ready for forks.

Begin your research in Discord #custom-forks

Please do not ask about forks outside of that channel.

Please start with `commaai/master-ci`. This is the latest upstream and supports lateral only. 

> [!WARNING]

> > Some forks will brick your C3X.
> > Most forks are not made for C3X because C3X is less than a year old.

Reviewer: ```suggestion
> Some forks will brick your C3X. \
> Most forks are not made for C3X because C3X is less than a year old.
```

> 

> > Some forks contain banned code.
> > Using it will get you banned by the comma.ai company.

Reviewer: ```suggestion
> Some forks contain banned code. \
> Using it will get you banned by the comma.ai company.
```

> 

> > Some forks have nudgeless-lane-change.
> > Simply clicking the turn signal will move your car to the next lane.
> > Without any checks.
> > Yes, it will drive into the car next to you.

Reviewer: ```suggestion
> Some forks have nudgeless-lane-change. \
> Simply clicking the turn signal will move your car to the next lane. \
> Without any checks. \
> Yes, it will drive into the car next to you.
```

> 
> Some forks play a blood-curdling goat scream at max volume randomly.
> 
> The danger in using forks is not theoretical because it has full control over your car.
> 
> There's a real and present danger in using forks, even on the most popular forks.

If you acknowledge the warning above and are still looking to try a fork that supports TSK the following are available to install at your own risk. Please keep in mind that these are community maintained and may not stay up to date:

#### Forks with Long
* `alexandresato/personal3`
  * A.K.A SatoPilot, the first fork to support TSK long!

>   * Very quick S&G response

Reviewer: ```suggestion
  * Very quick stop-and-go response
```

>   * Includes TSK key extract button

Reviewer: Does it include it? I thought it was in a separate branch

Author: hmm I think you are right. I will have to double check.

* `chrispypatt/sunnypilot`
  * Chrispypatt's fork of SunnyPilot's in-progress rewrite. 
  * Maintained as best effort. This may not always be up to date with latest SunnyPilot. Reach out to @Chrispypatt for requests to rebase his fork with SunnyPilot's re-write if it is out of date.
  * Includes a TSK keyboard with key caching
  * Model switcher to easily switch between various models
#### Forks with Lateral Only
* `optskug/SiennaFP`

>   * Anbuck's old fork of FP and first fork to support TSK lateral!
> ---

Reviewer: ```suggestion
  * Anbuck's old fork of FP and first fork to support TSK lateral!
  * Uses an old AGNOS version. When downgrading, OP may get stuck in a registration loop. In this case, tap-tap-tap on the boot logo and reset the device to recover and then install again.
---
```

Go back to Step 4A-4 and then finish with Step 4A-5.

---

> ## Forks
> ### Which Fork Should I Use?
> > [!WARNING]

Reviewer: ```suggestion
## Forks

### Which Fork Should I Use?

> [!WARNING]
```

> If you have to ask, you're not ready for forks.

Begin your research in Discord #custom-forks

Please do not ask about forks outside of that channel.

Please start with `commaai/master-ci`. This is the latest upstream and supports lateral only. 

> [!WARNING]
> Some forks will brick your C3X. \
> Most forks are not made for C3X because C3X is less than a year old.
> 
> Some forks contain banned code. \
> Using it will get you banned by the comma.ai company.
> 
> Some forks have nudgeless-lane-change. \
> Simply clicking the turn signal will move your car to the next lane. \
> Without any checks. \
> Yes, it will drive into the car next to you.
> 
> Some forks play a blood-curdling goat scream at max volume randomly.
> 
> The danger in using forks is not theoretical because it has full control over your car.
> 
> There's a real and present danger in using forks, even on the most popular forks.

If you acknowledge the warning above and are still looking to try a fork that supports TSK the following are available to install at your own risk. Please keep in mind that these are community maintained and may not stay up to date:

> #### Forks with Long
> * `alexandresato/personal3`

Reviewer: ```suggestion
#### Forks with Long

* `alexandresato/personal3`
```

  * A.K.A SatoPilot, the first fork to support TSK long!
  * Very quick stop-and-go response
  * Includes TSK key extract button
* `chrispypatt/sunnypilot`
  * Chrispypatt's fork of SunnyPilot's in-progress rewrite. 
  * Maintained as best effort. This may not always be up to date with latest SunnyPilot. Reach out to @Chrispypatt for requests to rebase his fork with SunnyPilot's re-write if it is out of date.
  * Includes a TSK keyboard with key caching

>   * Model switcher to easily switch between various models
> #### Forks with Lateral Only
> * `optskug/SiennaFP`

Reviewer: ```suggestion
  * Model switcher to easily switch between various models

#### Forks with Lateral Only

* `optskug/SiennaFP`
```

>   * Anbuck's old fork of FP and first fork to support TSK lateral!

Reviewer: ```suggestion
  * @anbuck's old fork of FP and first fork to support TSK lateral!
```

Reviewer: On discord his username is @anrum. Not sure which we should list there

>   * @anbuck's old fork of FP and first fork to support TSK lateral!

Reviewer: GitHub username is `anbuck` but Discord is `anrum`. I was hoping @anbuck would link to his GH profile but it doesn't seem to do so. History section uses Discord username, so maybe `anrum`? 🤔

Author: Yeah I'm in favor of just using discord handles. GH username is not as useful for getting in touch with someone. Plus we can always link to the GH repo which should give them a path to the GH profile

>   * Uses an old AGNOS version. When downgrading, OP may get stuck in a registration loop. In this case, tap-tap-tap on the boot logo and reset the device to recover and then install again.
> ---

Reviewer: ```suggestion
  * Uses an old AGNOS version. When downgrading, OP may get stuck in a registration loop. In this case, tap-tap-tap on the boot logo and reset the device to recover and then install again.

---
```

If you acknowledge the warning above and are still looking to try a fork that supports TSK the following are available to install at your own risk. Please keep in mind that these are community maintained and may not stay up to date:

#### Forks with Long

* `alexandresato/personal3`
  * A.K.A SatoPilot, the first fork to support TSK long!
  * Very quick stop-and-go response
  * Also `alexandresato/extract_secoc_key_btn` includes a TSK key extract button and is rebased with personal3 often.
* `chrispypatt/sunnypilot`
  * Chrispypatt's fork of SunnyPilot's in-progress rewrite. 
  * Maintained as best effort. This may not always be up to date with latest SunnyPilot. Reach out to @Chrispypatt for requests to rebase his fork with SunnyPilot's rewrite.
  * Includes a TSK keyboard with key caching
  * Model switcher to easily switch between various models

#### Forks with Lateral Only

* `optskug/SiennaFP`
  * anrum's old fork of FP and first fork to support TSK lateral!
  * Uses an old AGNOS version. When downgrading, OP may get stuck in a registration loop. In this case, tap-tap-tap on the boot logo and reset the device to recover and then install again.

>   > [!WARNING]

Reviewer: ```suggestion

  > [!WARNING]
```
Ah poop, it's not rendering correctly. Let me try this one instead

If you acknowledge the warning above and are still looking to try a fork that supports TSK the following are available to install at your own risk. Please keep in mind that these are community maintained and may not stay up to date:

#### Forks with Long

* `alexandresato/personal3`
  * A.K.A SatoPilot, the first fork to support TSK long!
  * Very quick stop-and-go response
  * Also `alexandresato/extract_secoc_key_btn` includes a TSK key extract button and is rebased with personal3 often.
* `chrispypatt/sunnypilot`
  * Chrispypatt's fork of SunnyPilot's in-progress rewrite. 
  * Maintained as best effort. This may not always be up to date with latest SunnyPilot. Reach out to @Chrispypatt for requests to rebase his fork with SunnyPilot's rewrite.
  * Includes a TSK keyboard with key caching
  * Model switcher to easily switch between various models

#### Forks with Lateral Only

* `optskug/SiennaFP`
  * anrum's old fork of FP and first fork to support TSK lateral!
  * Uses an old AGNOS version. When downgrading, OP may get stuck in a registration loop. In this case, tap-tap-tap on the boot logo and reset the device to recover and then install again.

> 
>   > [!WARNING]
>   > `optskug/SiennaFP` is the only FrogPilot fork/branch with TSK support. Do not install the latest FP unless FrogGoesMoo confirmed that it supports TSK vehicles.

Reviewer: ```suggestion

> [!WARNING]
> `optskug/SiennaFP` is the only FrogPilot fork/branch with TSK support. Do not install the latest FP unless FrogGoesMoo confirmed that it supports TSK vehicles.
```
